src/framework/utils.hpp

```diff
@@ -335,7 +335,143 @@ cmatrix_t Matrix::U3(double theta, double phi, double lambda) {
   return mat;
 }
 
+//------------------------------------------------------------------------------
+// Static Vectorized Matrices
+//------------------------------------------------------------------------------
+class VMatrix {
+public:
+  // Single-qubit gates
+  const static cvector_t I;     // name: "id"
+  const static cvector_t X;     // name: "x"
+  const static cvector_t Y;     // name: "y"
+  const static cvector_t Z;     // name: "z"
+  const static cvector_t H;     // name: "h"
+  const static cvector_t S;     // name: "s"
+  const static cvector_t Sdg;   // name: "sdg"
+  const static cvector_t T;     // name: "t"
+  const static cvector_t Tdg;   // name: "tdg"
+  const static cvector_t X90;   // name: "x90"
+
+  // Two-qubit gates
+  const static cvector_t CX;    // name: "cx"
+  const static cvector_t CZ;    // name: "cz"
+  const static cvector_t SWAP;  // name: "swap"
+  const static cvector_t CR;    // TODO
+  const static cvector_t CR90;  // TODO
+
+  // Identity Matrix
+  static cvector_t Identity(size_t dim);
+
+  // Single-qubit waltz gates
+  static cvector_t U1(double lam);
+  static cvector_t U2(double phi, double lam);
+  static cvector_t U3(double theta, double phi, double lam);
+
+  // Complex arguments are implemented by taking std::real
+  // of the input
+  inline static cvector_t U1(complex_t lam) {return U1(std::real(lam));}
```

Comment:
`const` and lower case. There's no need to specify `inline` here as we are in a header file, so this is always inlined.
```c++
static cvector_t u1(...) const;
```